### DISCUSSION TITLE
Futexes in xskdev for tx_lock instread of pthread mutex

### DISCUSSION THREAD

**Discussion by maryamtahhan** [CloudNativeDataPlane/cndp#270] [General]
Hi Folks

Just wondering is it worth looking at using a futex2 for tx_lock in xskdev instead of the pthread mutex for lower overhead/better performance?

**Comment by KeithWiles** [comment]
If we can reduce the overhead I think it is a reasonable idea. Does futex2 have any dependencies i.e., a specific OS or version?
